constellations/gemini-admin/src/home/home.js

```diff
@@ -2,107 +2,151 @@
 require("./home.sass");
 import React from "react";
 import API from "../api.js";
-import { Link } from "react-router-dom";
+import { HomeTabSectionClasses } from "./homeClasses.js";
+import { HomeTabSectionUsers } from "./homeUsers.js";
+import { HomeTabSectionRegistrations } from "./homeRegistrations.js";
+import { HomeTabSectionAccounts } from "./homeAccounts.js";
+
+const sectionDisplayNames = {
+    class: "Unpublished Classes",
+    user: "New Users",
+    registration: "New Registrations",
+    unpaid: "Unpaid Accounts",
+};
 
 export class HomePage extends React.Component {
     state = {
-        classes: [],
+        currentSection: "class",
+
+        unpublishedClasses: [],
+        newUsers: [],
+        newUserClasses: [],
+        newUserAfh: [],
+        unpaidAccounts: [],
     };
 
-    componentDidMount() {
-        this.fetchData();
-    }
+    changeSection = (sectionName) => {
+        this.setState({
+            currentSection: sectionName,
+        });
+    };
 
-    fetchData = () => {
+    componentDidMount = () => {
         API.get("api/unpublished").then((res) => {
             const unpublishedList = res.data;
             this.setState({
-                classes: unpublishedList.classes,
+                unpublishedClasses: unpublishedList.classes,
+            });
+        });
+
+        API.get("api/users/new").then((res) => {
+            const users = res.data;
+            this.setState({
+                newUsers: users,
+            });
+        });
+
+        //pending registration for classes
+        API.get("api/user-classes/new").then((res) => {
+            const userClass = res.data;
+            this.setState({
+                newUserClasses: userClass,
+            });
+        });
+
+        //afh registration
+        API.get("api/userafhs/new").then((res) => {
+            const userAfh = res.data;
+            this.setState({
+                newUserAfh: userAfh,
+            });
+        });
+
+        API.get("api/accounts/unpaid").then((res) => {
+            const accounts = res.data;
+            this.setState({
+                unpaidAccounts: accounts,
             });
         });
     };
```

Comment:
Instead of multiple separate API calls, you could use `axios` to combine them into one. This way, you can have only one catch block. It would look something like this:
```
const apiCalls = [
    API.get("api/unpublished"),
    API.get("api/users/new"),
    // etc
];

axios
    .all(apiCalls)
    .then(
        axios.spread((...responses) => {
            this.setState({
                unpublishedClasses: responses[0].data.classes,
                newUsers: responses[1].data,
                // etc
            })
        })
    )
    .catch((err) => {
        console.log("Error: api call failed. " + err.message);
    });
```